sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/ModuleIdentityLifecycleSamples.cs

```diff
@@ -120,34 +119,149 @@ public async Task<ModuleIdentity> CreateModuleIdentityAsync(DeviceIdentity devic
+            try
+            {
+                Console.WriteLine($"Listing all modules in device with Id: '{deviceIdentity.DeviceId}'\n");
+
+                Response<IReadOnlyList<ModuleIdentity>> response = await IoTHubServiceClient.Modules.GetIdentitiesAsync(deviceIdentity.DeviceId);
+
+                foreach (ModuleIdentity moduleIdentity in response.Value)
+                {
+                    SampleLogger.PrintSuccess($"\t- Module Id: '{moduleIdentity.ModuleId}', Device Id: '{moduleIdentity.DeviceId}', ETag: '{moduleIdentity.Etag}'");
+                }
+
+                return response.Value;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to list module identities due to:\n{ex}");
+                throw;
+            }
         }
 
-        public Task UpdateModuleIdentityAsync()
+        /// <summary>
+        /// Get a module identity
+        /// </summary>
+        /// <param name="deviceIdentity">The device identity module belongs to.</param>
+        /// <param name="moduleId">The unique identifier of the module to get.</param>
+        public async Task<ModuleIdentity> GetModuleIdentityAsync(DeviceIdentity deviceIdentity, string moduleId)
         {
-            return Task.CompletedTask;
-        }
+            SampleLogger.PrintHeader("GET A MODULE");
 
-        public Task GetModuleTwinAsync()
+            try
+            {
+                Console.WriteLine($"Getting module identity with Id: '{moduleId}'\n");
+
+                Response<ModuleIdentity> response = await IoTHubServiceClient.Modules.GetIdentityAsync(deviceIdentity.DeviceId, moduleId);
+
+                ModuleIdentity moduleIdentity = response.Value;
+
+                SampleLogger.PrintSuccess($"\t- Module Id: '{moduleIdentity.ModuleId}', Device Id: '{moduleIdentity.DeviceId}', ETag: '{moduleIdentity.Etag}'");
+
+                return moduleIdentity;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to get a module identity due to:\n{ex}");
+                throw;
+            }
+        }
+        
+        /// <summary>
+        /// Update a module identity
+        /// </summary>
+        /// <param name="moduleIdentity">Module identity to be updated.</param>
+        public async Task<ModuleIdentity> UpdateModuleIdentityAsync(ModuleIdentity moduleIdentity)
         {
-            return Task.CompletedTask;
+            SampleLogger.PrintHeader("UPDATE A MODULE");
+
+            try
+            {
+                Console.WriteLine($"Updating module identity with Id: '{moduleIdentity.ModuleId}'. Setting 'ManagedBy' property to: '{Environment.UserName}'\n");
+                moduleIdentity.ManagedBy = Environment.UserName;
+
+                Response<ModuleIdentity> response = await IoTHubServiceClient.Modules.CreateOrUpdateIdentityAsync(moduleIdentity, IfMatchPrecondition.UnconditionalIfMatch);
```

Comment:
nit - the explanation for `IfMatchPrecondition.UnconditionalIfMatch` would be available as a tooltip, but it wouldn't hurt to add that as a note in here as well.

Reply:
definitely. I am still waiting for the service to come back with a response before I take action on all if-match related code.